gobblin-modules/gobblin-helix/src/main/java/gobblin/metastore/ZkStateStore.java

```diff
@@ -321,4 +365,44 @@ private void deserialize(byte[] data, List<T> states, String stateId) throws IOE
   private void deserialize(byte[] data, List<T> states) throws IOException {
     deserialize(data, states, null);
   }
-}
\ No newline at end of file
+
+  private boolean isCurrent(String tableName) {
+    StateStoreTableInfo tableInfo = StateStoreTableInfo.get(tableName);
+    return tableInfo.isCurrent();
+  }
+
+  private String getTablePath(String storeName, String tableName) throws IOException {
+    StateStoreTableInfo tableInfo = StateStoreTableInfo.get(tableName);
+    if (tableInfo.isCurrent()) {
+      return getLatestTablePath(storeName, tableInfo.getPrefix());
+    }
+    return formPath(storeName, tableName);
+  }
+
+  private String getLatestTablePath(String storeName, final String tableNamePrefix) throws IOException {
+    String path = Paths.get("/", storeName).toString();
+    List<String> children = propStore.getChildNames(path, 0);
+
+    if (children == null) {
+      return null;
+    }
+
+    String latestStatePath = null;
+    for (String c : children) {
+      if ((Strings.isNullOrEmpty(tableNamePrefix) ||
+        c.startsWith(tableNamePrefix + StateStoreTableInfo.TABLE_PREFIX_SEPARATOR))) {
+        if (latestStatePath == null) {
+          log.debug("Latest table for {}/{}* set to {}", storeName, tableNamePrefix, c);
+          latestStatePath = c;
+        } else if (c.compareTo(latestStatePath) > 0) {
```

Comment:
@htran1 I also think I have a reasonable solution for this.  Given we know the Urn, we can strip that off the front of the `tableName`, remove the extension, and then parse the remaining with the well known Id format.  Then we can directly compare the sequence numbers to see which is the latest.